feat(segment): add anonymous id sync

DavidWells opened the pull request:
Fix for https://github.com/DavidWells/analytics/issues/11

@tommedema this will automatically sync the anon id from segment with analytics anon id.

I'm wondering if we want to make this an option (with the default being on)

Trying to work through different use cases in my head.

What do you think?

DavidWells committed:
feat(segment): add anonymous id sync

tommedema commented:
- agree that this should be an option (default on tho that would be a breaking change so major semver), in case people get conflicting anonymous IDs from various integrations (segment, mixpanel, etc.); you could do it default off in this major version and then default on in the next major

- implementation seems solid

DavidWells committed:
chore: add examples installer script

DavidWells committed:
feat(segment): add syncAnonymousId setting

syncAnonymousId will automatically set segment anonymous Id to the analytics anon id after analytics bootstraps

DavidWells commented:
Named the setting `syncAnonymousId` 

Example usage:

```js
import Analytics from 'analytics'
import segmentPlugin from 'analytics-plugin-segment'

const analytics = Analytics({
  debug: true,
  plugins: [
	segmentPlugin({
	 writeKey: 'your-key',
	 syncAnonymousId: true,
	})
  ]
})
```
